planner/core/optimizer.go

```diff
@@ -382,6 +382,7 @@ func postOptimize(sctx sessionctx.Context, plan PhysicalPlan) PhysicalPlan {
 	plan = enableParallelApply(sctx, plan)
 	handleFineGrainedShuffle(sctx, plan)
 	checkPlanCacheable(sctx, plan)
+	propagateProbeParents(plan, nil)
```

Comment:
Should we invoke this function only when `sctx.GetSessionVars().StmtCtx.InExplainStmt == true`?

Reply:
No.
Because of features like slow query and statements summary, every execution plan is possible to be recorded and displayed.